### PR TITLE
STM32: Board conversion to DT clock configuration. STM32F4 part2

### PR DESCRIPTION
**Pull request by ABOSTM**
Following #34609, Board conversion to DT clock configuration. STM32L4 boards, part2.
Compilation OK. Not tested (I don't have the boards).

* nucleo_f410rb
* nucleo_f412zg
* nucleo_f413zh
* nucleo_f446re
* segger_trb_stm32f407
* adafruit_feather_stm32f405
* steval_fcu001v1
* stm32f4_disco
* stm32f411e_disco
* stm32f412g_disco
* stm32f429i_disc1
* stm32f469i_disco

### CODE REVIEW

File: boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts [hunk 1]
```diff
@@ -34,6 +34,28 @@
...
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(168)>;
+	ahb-prescaler = <1>;
+	apb1-prescaler = <4>;
+	apb2-prescaler = <1>;
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts:56]: apb2-prescaler = <2>;

File: boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig [hunk 2]
```diff
@@ -21,16 +18,9 @@ CONFIG_PINMUX=y
...
-# SYSCLK selection
-CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL=y
-
-# use HSE as PLL input
-CONFIG_CLOCK_STM32_PLL_SRC_HSE=y
-CONFIG_CLOCK_STM32_HSE_CLOCK=16000000
-
 # produce 84MHz clock at PLL output
 CONFIG_CLOCK_STM32_PLL_M_DIVISOR=16
 CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=336
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig:26]: These lines could be removed now.

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
CI failing due to:
```
tests/arch/arm/arm_no_multithreading/arch.arm.no_multithreading in segger_trb_stm32f407:tests/arch/arm/arm_no_multithreading/arch.arm
```

File: boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts (hunk 1 above)
Comment (ABOSTM) [reply boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts]: Right. Done.

File: boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig (hunk 2 above)
Comment (ABOSTM) [reply boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig]: Right. Done

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**